src/Tools/dotnet-trace/CommandLine/Commands/CollectCommand.cs

```diff
@@ -148,32 +149,43 @@ private static async Task<int> Collect(CancellationToken ct, IConsole console, i
 
                             using (var fs = new FileStream(output.FullName, FileMode.Create, FileAccess.Write))
                             {
-                                Console.Out.WriteLine($"Process        : {process.MainModule.FileName}");
-                                Console.Out.WriteLine($"Output File    : {fs.Name}");
+                                systemConsole.Out.WriteLine($"Process        : {process.MainModule.FileName}");
+                                systemConsole.Out.WriteLine($"Output File    : {fs.Name}");
                                 if (shouldStopAfterDuration)
-                                    Console.Out.WriteLine($"Trace Duration : {duration.ToString(@"dd\:hh\:mm\:ss")}");
+                                    systemConsole.Out.WriteLine($"Trace Duration : {duration.ToString(@"dd\:hh\:mm\:ss")}");
+
+                                systemConsole.Out.WriteLine("\n");
+
+                                var region = new Region(0, systemConsole.CursorTop, 55, 1, true);
+                                var renderer = new ConsoleRenderer(systemConsole, resetAfterRender: true);
+                                systemConsole.Out.WriteLine("\nPress <Enter> or <Ctrl+C> to exit...");
 
-                                Console.Out.WriteLine("\n\n");
                                 var buffer = new byte[16 * 1024];
 
+                                Task.Run(() =>
+                                {
+                                    while (true)
+                                    {
+                                        if (!fs.CanWrite)
+                                            break;
+                                        renderer.RenderToRegion($"[{stopwatch.Elapsed.ToString(@"dd\:hh\:mm\:ss")}]\tRecording trace {GetSize(fs.Length)}", region);
```

Comment:
By default all non-static methods on .NET framework APIs are not thread-safe. Although I suspect it winds up working you are taking a dependency on undefined/implementation-specific behavior when you manipulate the FileStream API from different tasks that are likely running on different threads. A safe way to do this would be to define a ViewModel object that has whatever state you need to render in the UI, update it in the buffer processing loop, read it here, and then use synchronization primitives like a lock if you need any guarantees that the reads/writes are atomic.

Reply:
Good thinking.  I'll refactor to something with fewer cross-thread concerns.